geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSet.java

```diff
@@ -340,7 +306,7 @@ long srem(ArrayList<ByteArrayWrapper> membersToRemove, Region<RedisKey, RedisDat
    * @return a set containing all the members in this set
    */
   @VisibleForTesting
-  Set<ByteArrayWrapper> smembers() {
+  public Set<byte[]> smembers() {
     return new HashSet<>(members);
```

Comment:
This definitely needs to be an Object2ObjectOpenCustomHashMap, not a HashMap here.

Reply:
I would actually suggest changing this to `Collections.unmodifiableSet(members)` here. This is an expensive operation, so it would be better if we only make a copy of this set where we actually need to modify it. Or only currently supported command, SMEMBERS, doesn't need this extra copying.

Reply:
The commit prior to the current head returned an `ObjectOpenCustomHashSet` here (and in NullRedisSet) but this caused serialization issues in DUnit tests as that class wasn't allowed by the serialization filter. After trying to find an alternate solution, I've settled on extending `ObjectOpenCustomHashSet` to allow adding it to the appropriate sanctioned-serializables file, and to allow proper support of `addAll()` and `removeAll()`, since the current implementation of `ObjectOpenCustomHashSet` doesn't employ the hashing strategy set in the constructor for those.

Reply:
Correction, the hashing strategy is correctly employed in the current implementation, it's just the serialization issue that's present. Nevertheless, extending the class allows serialization and accurate calculation of size down the line when we revisit sizeable for RedisSet, so it's still the best choice, I think.